### ISSUE TITLE
nitrocli's name

### ISSUE THREAD

**Issue by jans23** [d-e-s-o/nitrocli#13]
First of all, thank you very much for this very helpful piece of software. We at Nitrokey depend on exactly such initiatives to get the overall project moving.

I'm wondering if the name `nitrocli` is the right name, especially when we anticipate that this software may become available in Linux distributions (as discussed in a separate issue ticket). I doubt because I find `nitrocli`not very expressive. Perhaps something like `nitrokey-cli` or  `nitrokey-tool`would be more self explanatory?

**Comment by robinkrahl** [comment]
I don’t think the name is that important.  Most users will probably use a search engine or look at the Nitrokey documentation – they will find the tool with any name.  Even people searching within the package manager should be able to find it.  For example, `apt search` also searches the package description which will of course contain the term Nitrokey.

What I like about the current name is that it is unique and concise at the same time.  I must admit that nitrocli does not show up on the first page when googling “nitrokey cli” or “nitrokey command line”.  But your blog entries and Github issues do, so once there is a stable release and if you mention the tool on the top-ranked sites, the visiblity should improve.

So in conclusion:  I do like the current name, but if you think that a more verbose name would benefit the tool, go ahead.

**Comment by d-e-s-o** [comment]
Thanks for bringing this point up @jans23 . I am obviously biased but I also like the name. In fact, it's the one thing I am fairly proud of about this project :P

Now most of this is clearly a subjective matter, but I do agree with @robinkrahl that if your main concern evolves around discoverability, that the name should not be the major factor to consider for the reasons already stated.
There are a lot of arguably successful projects out there whose name reveals fairly little about the type of functionality they provide, consider for example [`rocket`](https://crates.io/crates/rocket), [`servo`](https://servo.org/) from the Rust ecosystem, [`Opera`](https://www.opera.com/), [`Kubernetes`](https://kubernetes.io/), or even more traditional utilities like `awk`. While initially opposed to that, I've taken almost a 180° turn on this front.

Anyway, I believe that `nitrocli` provides a nice middle ground in this regard in that it is close enough to the hardware device it works with to evoke associations while hopefully being different enough (even in speech) to not cause confusion. At the same time it is concise, expresses its intent to the user understanding what a `CLI` is, and avoids the need for hyphens (which I've been trying to avoid hard in names in recent times).

That is a long-winded way of saying that I'd really prefer `nitrocli` over the proposed `nitrokey-cli` or `nitrokey-tool`. Now we can certainly think about this aspect some more and evaluate other proposals (in case there are some), but ultimately I am fairly happy and we would need to find a good candidate to seriously consider a rename.

**Comment by jans23** [comment]
I don't have other proposals than `nitrokey-cli` and `nitrokey-tool` and I understand these were rejected (which is ok for me). :-) 
Also, how to pronounce the project name? Is it "Nitro C L I" or "Nitrokli"?

**Comment by d-e-s-o** [comment]
> Also, how to pronounce the project name? Is it "Nitro C L I" or "Nitrokli"?

My intent was to go with "Nitrokli" (which, it is my understanding, is a reasonably common way to go about the "cli" part).

> I don't have other proposals than nitrokey-cli and nitrokey-tool and I understand these were rejected (which is ok for me). :-)

Okay, thanks for your understanding. I don't see any action pending on this issue then and will go ahead and close it.
